synapse/lib/parser.py

```diff
@@ -425,17 +438,12 @@ def _larkToSynExc(self, e):
         at = len(self.text)
         line = None
         column = None
-        if isinstance(e, lark.exceptions.UnexpectedCharacters):
-            expected = sorted(terminalEnglishMap[t] for t in e.allowed)
-            mesg += f'.  Expecting one of: {", ".join(expected)}'
+        if isinstance(e, lark.exceptions.UnexpectedToken):
+            expected = sorted(terminalEnglishMap[t] for t in e.expected)
+            mesg += f'  Expecting one of: {", ".join(expected)}'
             at = e.pos_in_stream
             line = e.line
             column = e.column
-        elif isinstance(e, lark.exceptions.UnexpectedEOF):
```

Comment:
was EOF error removed? Or is that no longer able to be raised with LALR ?

Reply:
All the instances of EOF i've seen raise an `Unexpected token Token('$END'` instead of UnexpectedEOF, though I did ask the Lark guys about this and apparently it is still possible for LALR to raise an UnexpectedEOF so I will add this back in just in case.